src/modals/ConnectModal.svelte

```diff
@@ -0,0 +1,180 @@
+<script lang="ts">
+  import * as httpd from "@app/lib/httpd";
+  import * as modal from "@app/lib/modal";
+  import { httpdStore } from "@app/lib/httpd";
+
+  import Command from "@app/components/Command.svelte";
+  import FullscreenModal from "@app/components/FullscreenModal.svelte";
+  import Icon from "@app/components/Icon.svelte";
+  import IconButton from "@app/components/IconButton.svelte";
+  import IconSmall from "@app/components/IconSmall.svelte";
+  import Link from "@app/components/Link.svelte";
+  import TextInput from "@app/components/TextInput.svelte";
+
+  export let title: string;
+  export let subtitle: string[];
+
+  $: customUrl = `${httpd.api.baseUrl.scheme}://${httpd.api.baseUrl.hostname}:${customPort}`;
+  $: command = import.meta.env.PROD
+    ? `rad web --backend ${customUrl}`
+    : `rad web --frontend ${
+        new URL(import.meta.url).origin
+      } --backend ${customUrl}`;
+  let customPort = httpd.api.port;
+  $: validPortNumber = Number(customPort) > 0 && Number(customPort) <= 65535;
+
+  $: if ($httpdStore.state === "authenticated") {
+    modal.hide();
+  }
+</script>
+
+<style>
+  .container {
+    display: flex;
+    flex-direction: column;
+    width: 33rem;
+    margin-top: 1.5rem;
+    gap: 1.5rem;
+  }
+  .progress {
+    display: flex;
+    flex-direction: column;
+    gap: 0.5rem;
+  }
+  .progress-bar {
+    height: 6px;
+    border-radius: var(--border-radius-round);
+    background-color: var(--color-background-dip);
+  }
+  .bar {
+    display: flex;
+    background-color: var(--color-fill-secondary);
+    height: 100%;
+    border-radius: var(--border-radius-round);
+  }
+  .captions {
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr;
+    color: var(--color-foreground-dim);
+    font-size: var(--font-size-tiny);
+  }
+
+  .input {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    width: 100%;
```

Comment:
Not needed

Reply:
591e4f1a7fcc2f9a0314e9bd1d26c92f1973288b